materialize-snowflake/snowflake.go

```diff
@@ -717,113 +717,105 @@ func (d *transactor) Acknowledge(ctx context.Context) (*pf.ConnectorState, error
 	// Keep asking for a report on the files that have been submitted for processing
 	// until they have all been successful, or an error has been thrown
 
-	// If we see no results from the REST API for `maxTries` iterations, then we
-	// fallback to asking the `COPY_HISTORY` table. We allow up to 5 minutes for results to show up in the REST API.
-	if len(pipes) > 0 {
-		// 5 minutes divided by three seconds for each pipe
-		var maxTries = (5 * 60) / (3 * len(pipes))
-		var retryDelay = 3 * time.Second
-		for tries := 0; tries < maxTries; tries++ {
-			for pipeName, pipe := range pipes {
-				// if the pipe has no files to begin with, just skip it
-				// we might have processed all files of this pipe previously
-				if len(pipe.files) == 0 {
-					continue
-				}
-
-				// We first try to check the status of pipes using the REST API's insertReport
-				// The REST API does not wake up the warehouse, hence our preference
-				// however this API is not the most ergonomic and sometimes does not yield
-				// results as expected
-				report, err := d.pipeClient.InsertReport(pipeName)
-				if err != nil {
-					return nil, fmt.Errorf("snowpipe: insertReports: %w", err)
-				}
-
-				// If the files have already been loaded, when we submit a request to
-				// load those files again, our request will not show up in reports.
-				// Moreover, insertReport only retains events for 10 minutes.
-				// One way to find out whether the files were successfully loaded is to check
-				// the COPY_HISTORY to make sure they are there. The COPY_HISTORY is much more
-				// reliable. If they are not there, then something is wrong.
-				if len(report.Files) == 0 {
-					// We try `maxTries` times since it may take some time for the REST API
-					// to reflect the new pipe requests
-					if tries < maxTries-1 {
-						time.Sleep(retryDelay)
-						continue
-					}
+	// If we see no results from the REST API for 5 minutes, then we fallback to asking the `COPY_HISTORY` table
+	var retryDelay = 500 * time.Millisecond
+	var maxTryTime = time.Now().Add(5 * time.Minute)
+
+	for len(pipes) > 0 {
+		for pipeName, pipe := range pipes {
+			// if the pipe has no files to begin with, just skip it
+			// we might have processed all files of this pipe previously
+			if len(pipe.files) == 0 {
+				delete(pipes, pipeName)
```

Comment:
Once a pipe has no more files to process, delete it from the map so the outer loop stops eventually